client/src/api.d.ts

```diff
@@ -0,0 +1,106 @@
+import type { CID } from "multiformats/cid"
+
+export interface Service {
+  endpoint: URL
+  token: string
+}
+
+export interface API {
+  /**
+   * Stores a single file and returns a corresponding CID.
+   */
+  storeBlob(service: Service, content: Blob|File): Promise<CID>
+  /**
+   * Stores a directory of files and returns a CID. Provided files **MUST**
+   * be within a same directory, otherwise error is raised. E.g. `foo/bar.png`,
+   * `foo/bla/baz.json` is ok but `foo/bar.png`, `bla/baz.json` is not.
+   */
+  storeDirectory(service: Service, files: Iterable<File>): Promise<CID>
+  /**
+   * Returns current status of the stored content by its CID.
+   */
+  status(service: Service, cid: CID): Promise<StatusResult>
+  /**
+   * Removes stored content by its CID from the service. Please note that
+   * even if content is removed from the service other nodes that have
+   * replicated it might still continue providing it.
+   */
+  delete(service: Service, cid: CID): Promise<void>
+}
+
+export interface StatusResult {
+  cid: CID
+  deals: Deals,
+  pin: Pin
+  created: Date
+}
+
+export type Deals = OngoingDeals | FinalizedDeals
```

Comment:
What is the reason for the distinction?

Reply:
Broadly speaking it given a simple way for user to tell if this is in progress or done ? More nuanced answer is ongoing deals are not finalized and deals array may change over time, deals added removed, their states update etc... Once it’s finalized it’s done and all deals are `FinalizedDeal` that specific fields that may not be present when in `OngoingDeals` state.